data-bulletproofing.md

```diff
@@ -69,6 +69,13 @@ _Russell Clemmings of the Fresno Bee on rechecking your data:_
 - Have someone who knows the data check your results before publication -- even the target of the story, if possible.
 - Double-check surprising results -- if citations spiked by 50 percent in one year, it could be a story or it could (more likely) be an error. 
 
+_Peter Harkins, formerly of the Washington Post on structuring data:_
+
+- Extract real data (choose the worst, noisiest stuff) to use as fixtures in automated tests and you'll know you haven't broken things as you go.
+- Don't model anything as many-to-many. Name the intermediate concept and give it two one-to-many associations. It will almost always want to accumulate more info in the future, and the cost of changing your model will be high.
```

Comment:
It will have to have an intermediate table, but I'm saying that intermediate table must be considered an entity in its own right. For instance, if you link `authors` and `screenplays` in Rails it'll call the join table `authors_screenplays`. Don't do that, name it `authorships`, model it, and treat it as a first-class record rather than plumbing. It will certainly gain complexity over time, and the longer you go before recognizing the relationships as a record, the more pain your code will suffer.